hmpps-prisoner-search-indexer/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/indexer/services/events/HmppsDomainEventsEmitterIntTest.kt

```diff
@@ -0,0 +1,572 @@
+    assertThatJson(message.Message).node("occurredAt").isEqualTo("2022-09-16T11:40:34+01:00")
+    assertThatJson(message.Message).node("detailUrl").isEqualTo("http://localhost:8080/prisoner/some_offender")
+    assertThatJson(message.Message).node("additionalInformation.nomsNumber").isEqualTo("some_offender")
+    assertThatJson(message.Message).node("additionalInformation.prisonId").isEqualTo("MDI")
+    assertThatJson(message.Message).node("additionalInformation.reason").isEqualTo("TRANSFERRED")
+  }
+
+  @Test
+  fun `e2e - will send prisoner updated event to the domain topic`() {
+    recreatePrisoner(PrisonerBuilder(prisonerNumber = "A1239DD", bookingId = null))
+
+    // update the prisoner on ES
+    prisonApi.stubFor(
+      get(urlEqualTo("/api/offenders/A1239DD"))
+        .willReturn(
+          aResponse()
+            .withHeader("Content-Type", "application/json")
+            .withBody(
+              PrisonerBuilder(
+                prisonerNumber = "A1239DD",
+                bookingId = null,
+                firstName = "NEW_NAME",
+              ).toOffenderBooking(),
+            ),
+        ),
+    )
+    val message = "/messages/offenderDetailsChanged.json".readResourceAsText().replace("A7089FD", "A1239DD")
+
+    offenderQueueSqsClient.sendMessage(message)
+    await untilCallTo { getNumberOfMessagesCurrentlyOnEventQueue() } matches { it == 0 }
+    await untilCallTo { getNumberOfMessagesCurrentlyOnDomainQueue() } matches { it == 1 }
+
+    val updateMsgBody = readNextDomainEventMessage()
+    assertThatJson(updateMsgBody).node("eventType").isEqualTo("test.prisoner-offender-search.prisoner.updated")
+    assertThatJson(updateMsgBody).node("additionalInformation.nomsNumber").isEqualTo("A1239DD")
+    assertThatJson(updateMsgBody).node("additionalInformation.categoriesChanged").isArray.containsExactlyInAnyOrder("PERSONAL_DETAILS")
+  }
+
+  @Test
+  fun `e2e - will send prisoner release event to the domain topic`() {
+    recreatePrisoner(PrisonerBuilder(prisonerNumber = "A1239DD", bookingId = null))
+
+    // update the prisoner on ES
+    prisonApi.stubFor(
+      get(urlEqualTo("/api/offenders/A1239DD"))
+        .willReturn(
+          aResponse()
+            .withHeader("Content-Type", "application/json")
+            .withBody(
+              PrisonerBuilder(
+                prisonerNumber = "A1239DD",
+                bookingId = null,
+                released = true,
+              ).toOffenderBooking(),
+            ),
+        ),
+    )
+    offenderQueueSqsClient.sendMessage(
+      "/messages/offenderDetailsChanged.json".readResourceAsText().replace("A7089FD", "A1239DD"),
+    )
+    await untilCallTo { getNumberOfMessagesCurrentlyOnDomainQueue() } matches { it == 2 }
+    val nextTwoEventTypes = listOf(readEventFromNextDomainEventMessage(), readEventFromNextDomainEventMessage())
+
+    assertThat(nextTwoEventTypes).containsExactlyInAnyOrder(
```

Comment:
some of these tests appeared to be a bit flaky, so have amended them to allow the two events to be read in any order